File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        '''
        track key: frequency of each letter (o(26) for 26 letters) and value: group list

        if frequescy array in seen, add to group
        else add key value to seen 
        '''

        ans = {}
        for w in strs:
            track = [0]*26
            for i in w:
                track[ord(i) - 97] += 1
            track = tuple(track)
            if track in ans:
                ans[track].append(w)
            else:
                ans[track] = [w]
        return [ans[i] for i in ans]

File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        // compare each letter of each word. if the letters dont match return the prefix 
        string prefix{""};
        for (int i{0}; i < strs[0].size(); i++) {
            char cur{strs[0][i]};
            for (string s : strs) {
                if (s[i] != cur) return prefix;
            }
            prefix += cur;
        }
        return prefix;
    }
};